actors/common/src/main/java/org/sunbird/learner/actors/skill/UserSkillManagementActor.java

```diff
@@ -3,68 +3,186 @@
+
+      Map<String, Object> userMap = new HashMap();
+      if (!searchedUserList.isEmpty()) {
+        userMap = searchedUserList.get(0);
+      }
+      ObjectMapper objectMapper = new ObjectMapper();
+      List<Skill> userSkills =
+              objectMapper.convertValue(
+                      userMap.get(JsonKey.SKILLS), new TypeReference<List<Skill>>() {
+                      });
+      HashSet<Skill> userSkillsSet = new HashSet<>(userSkills);
+      Iterator<String> skillListItr = skillList.iterator();
+      while (skillListItr.hasNext()) {
+        String skillName = skillListItr.next();
+        if (!StringUtils.isBlank(skillName)) {
+
+          // check whether user have already this skill or not -
+          String id =
+                  OneWayHashing.encryptVal(
+                          userId + JsonKey.PRIMARY_KEY_DELIMETER + skillName.toLowerCase());
+          Iterator<Skill> itr = userSkillsSet.iterator();
+          while (itr.hasNext()) {
+            Skill skill = itr.next();
+            if (skill.getId().equalsIgnoreCase(id)) {
+              skillListItr.remove();
+              itr.remove();
+            }
+          }
+        }
+      }
+
+      if (CollectionUtils.isNotEmpty(skillList)) {
+        saveUserSkill(skillList, userId);
+      }
+      if (CollectionUtils.isNotEmpty(userSkillsSet)) {
+        List<String> idList = userSkillsSet.stream().map(skill -> skill.getId()).collect(Collectors.toList());
+       Boolean deleted = userSkillDao.deleteUserSkill(idList);
+       if(!deleted){
+         ProjectLogger.log(
+                 "Delete Failed for " + userId,
+                 idList,
+                 LoggerEnum.ERROR.name());
+
+       }
+
+        updateEs(userId);
+      }
     }
+    Response response3 = new Response();
+    response3.getResult().put(JsonKey.RESULT, "SUCCESS");
+    sender().tell(response3, self());
+
+    targetObject =
+            TelemetryUtil.generateTargetObject(userId, JsonKey.USER, JsonKey.UPDATE, null);
+    TelemetryUtil.generateCorrelatedObject(userId, JsonKey.USER, null, correlatedObject);
+    TelemetryUtil.telemetryProcessingCall(
+            actorMessage.getRequest(), targetObject, correlatedObject);
+    updateSkillsList(skillList);
+
+
+  }
+
+
+  private void saveUserSkill(List<String> skillSet,String userId){
+    SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+    for (String skillName : skillSet) {
+      String id =
+              OneWayHashing.encryptVal(
+                      userId + JsonKey.PRIMARY_KEY_DELIMETER + skillName.toLowerCase());
+      Map<String, Object> userSkillMap = new HashMap<>();
+      userSkillMap.put(JsonKey.ID, id);
+      userSkillMap.put(JsonKey.USER_ID, userId);
+      userSkillMap.put(JsonKey.SKILL_NAME, skillName);
+      userSkillMap.put(JsonKey.SKILL_NAME_TO_LOWERCASE, skillName.toLowerCase());
+      userSkillMap.put(JsonKey.ADDED_BY, userId);
```

Comment:
Add following fields in user_skills table (similar to user_external_identity table). Migration of DB table needs to be taken care.

createdby text
createdon timestamp
lastupdatedby text
lastupdatedon timestamp

Drop following columns if not used by Sunbird mobile and portal.

addedat text
addedby text